pet-store-vanilla-js/app.js

```diff
@@ -0,0 +1,139 @@
+import {
+  configureFormNewModal,
+  configureFormEditModal,
+  showDeleteModal,
+} from './modals.js';
+import { getPetKinds, getAllPets } from './api.js';
+import { formatDate, hideError, showError } from './utils.js';
+import { formElements } from './form.js';
+
+const mainPageElements = {
+  addPetButton: document.getElementById('add-pet-btn'),
+  loadingPetsSpinner: document.getElementById('loading-pets-spinner'),
+  tableBody: document.getElementById('data-table').tBodies[0],
+};
+
+export const petKindsEnum = {};
+const allPets = [];
+window.addEventListener('DOMContentLoaded', async () => {
+  disableAddPetButton();
+  await refreshPets(true);
+  enableAddPetButton();
+});
+
+function disableAddPetButton() {
```

Comment:
Should be disabled by default. Also - use css to lower the opacity when the button is disabled - no need to do it manually.